GAE/test/org/akvo/flow/api/app/DataPointUtilTest.java

```diff
@@ -0,0 +1,190 @@
+ */
+
+package org.akvo.flow.api.app;
+
+import com.gallatinsystems.device.dao.DeviceDAO;
+import com.gallatinsystems.device.domain.Device;
+import com.gallatinsystems.framework.domain.BaseDomain;
+import com.gallatinsystems.survey.dao.SurveyDAO;
+import com.gallatinsystems.survey.domain.Survey;
+import com.gallatinsystems.surveyal.dao.SurveyedLocaleDao;
+import com.gallatinsystems.surveyal.domain.SurveyedLocale;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.akvo.flow.dao.SurveyAssignmentDao;
+import org.akvo.flow.domain.persistent.SurveyAssignment;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.waterforpeople.mapping.dao.SurveyInstanceDAO;
+import org.waterforpeople.mapping.domain.SurveyInstance;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DataPointUtilTest {
+    private final static LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    @BeforeAll
+    public static void setUp() {
+        helper.setUp();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        helper.tearDown();
+    }
+
+    private long randomId() {
+        return ThreadLocalRandom.current().nextLong(0, 1000000);
+    }
+
+    private int randomInt() {
+        return ThreadLocalRandom.current().nextInt(1, 200);
+    }
+
+    private Survey createForm() {
+        long id = randomId();
+        Survey form = new Survey();
+        form.setKey(KeyFactory.createKey("Survey", id));
+        form.setName("Form" + id);
+        form.setVersion(1.0d);
+        SurveyDAO dao = new SurveyDAO();
+        return dao.save(form);
+    }
+
+    private List<SurveyInstance> createFormInstances(Survey form, SurveyedLocale survey, int howMany) {
+        List<SurveyInstance> newInstances = new ArrayList<>();
+        for (int i = 0; i < howMany; i++) {
+            SurveyInstance si = new SurveyInstance();
+            si.setSurveyId(form.getKey().getId());
+            si.setSurveyedLocaleId(survey.getKey().getId());
+            si.setCollectionDate(new Date());
+            si.setUuid(UUID.randomUUID().toString());
+            si.setFormVersion(form.getVersion());
+            newInstances.add(si);
+        }
+        SurveyInstanceDAO dao = new SurveyInstanceDAO();
+        return (List<SurveyInstance>) dao.save(newInstances);
+    }
+
+    private List<SurveyInstance> createFormInstances(SurveyedLocale dataPoint, int howMany, Survey... forms) {
```

Comment:
What is homMany used for?

Reply:
Not sure if I understand the question, `howMany` for defining the number of entities to be created